src/xrpld/app/tx/detail/AMMDelete.cpp

```diff
@@ -72,8 +72,8 @@ AMMDelete::doApply()
     // as we go on processing transactions.
     Sandbox sb(&ctx_.view());
 
-    auto const ter =
-        deleteAMMAccount(sb, ctx_.tx[sfAsset], ctx_.tx[sfAsset2], j_);
+    auto const ter = deleteAMMAccount(
+        sb, ctx_.tx[sfAsset].get<Issue>(), ctx_.tx[sfAsset2].get<Issue>(), j_);
```

Comment:
What happens if someone submits an `AMMDelete` transaction with MPT assets? The `STAmount` fields have the protection of `soeMPTSupported`, but the `STIssue` fields do not, to my knowledge. Do they need it? I don't see a check that `tx[sfAsset].holds<Issue>` in `preflight`. This pattern is in all the AMM transactors, but I'm picking on `AMMDelete` just because it's the simplest. Unit tests that submit MPT assets would be good evidence that it is handled.

Reply:
`soeMPTSupported` is extended to `STIssue` [fields](https://github.com/gregtatcam/rippled/blob/feature/mpt-stissue/include/xrpl/protocol/SOTemplate.h#L75). The unit-tests are updated as [well](https://github.com/gregtatcam/rippled/blob/feature/mpt-stissue/src/test/app/MPToken_test.cpp#L1471).